src/flows/pipes/Downsample/view.tsx

```diff
@@ -20,23 +20,39 @@ import DurationInput from 'src/shared/components/DurationInput'
 
 import {PipeProp} from 'src/types/flows'
 
+const AVAILABLE_FUNCTIONS = FUNCTIONS.map(f => f.name)
+
 const Downsample: FC<PipeProp> = ({Context}) => {
   const {data, update} = useContext(PipeContext)
+  const options = useMemo(() => {
+    if (!data.functions || !data.functions.length) {
+      return [{name: 'mean'}].map(f => f.name)
+    }
+    return data.functions.map(f => f.name)
+  }, [data.functions])
+
+  const selectFn = useCallback(
+    (fn: string) => {
+      const fns = options.map(f => ({name: f}))
+      let found = false
+      let fnIdx = fns.findIndex(f => f.name === fn)
+
+      while (fnIdx !== -1) {
+        found = true
+        fns.splice(fnIdx, 1)
+        fnIdx = fns.findIndex(f => f.name === fn)
+      }
```

Comment:
wait so what's the point of this while loop? Seems like it should only run once?

Reply:
I'm nervous this will lead to an infinite loop

Reply:
its for when we have multiple instances of a function being populated into the functions list.. and that stuff gets persisted to local storage. it's like a bug that just wont go away.

do you have an irrational fear of loops?

Reply:
So like if `fn` isn't in `fns`, `fnIdx` will always be `-1` right? Are you saying that there is no chance of that happening? 

Yes I had a tragic early career experience regarding infinite loops that led to infinite-loop-ptsd 😆

Reply:
yes

it's gonna be alright buddy.. embrace the while loop AND the for loop. no function stack creation overhead when iterating on large data sets? yes PLEASE

Reply:
alright I got the boatie guarantee or my money back haha